ocaml/runtime/minor_gc.c

```diff
@@ -315,6 +318,16 @@ static void oldify_one (void* st_v, value v, volatile value *p)
         st->todo_list = v;
       } else {
         CAMLassert (sz == 1);
```

Comment:
The logic here took me a while to figure out. I don't think it's wrong, but I think it might be simpler to copy the non-scannable fields directly in oldify_one, as is currently done for entirely non-scannable blocks (where `tag >= No_scan-tag`).

That is, I think it would be simpler to directly copy the fields in `[scannable_sz, sz)`, and do the `sz > 1` tail-call test on `scannable_sz` instead (handling the `0` case with a return as you do here). That way, I think you can delete the new loop in `oldify_mopup`.

Reply:
Done! I like this structure more, too.